### PR TITLE
[TF] General echo cleanup

### PR DESCRIPTION
**Pull request by nmittler**
Splitting out structures into separate files and creating substructures in ConfigOptions.

**Please provide a description of this PR:**

### CODE REVIEW

File: pkg/test/framework/components/echo/calloptions.go [hunk 1]
```diff
@@ -24,6 +24,55 @@ import (
...
+	HTTP2 bool
+
+	// If true, HTTP/3 request over QUIC will be used.
+	// It is mandatory to specify TLS settings
+	HTTP3 bool
+
+	// Path specifies the URL path for the HTTP(s) request.
+	Path string
+
+	// Method to send. Defaults to HTTP.
```
Comment (howardjohn) [inline pkg/test/framework/components/echo/calloptions.go:39]:
```suggestion
	// Method to send. Defaults to GET.
```

?

File: pkg/test/framework/components/echo/calloptions.go [hunk 2]
```diff
@@ -24,6 +24,55 @@ import (
...
+	InsecureSkipVerify bool
+
+	Alpn       []string
+	ServerName string
+}
+
+// TCP settings
+type TCP struct {
+	// ExpectedResponse asserts this is in the response for TCP requests.
+	ExpectedResponse *wrappers.StringValue
```
Comment (howardjohn) [inline pkg/test/framework/components/echo/calloptions.go:73]: should probably be a checker but can do it later

File: pkg/test/framework/components/echo/calloptions.go [hunk 3]
```diff
@@ -39,71 +88,40 @@ type CallOptions struct {
...
-	// Skip verify peer's certificate.
-	InsecureSkipVerify bool
+	// TLS settings.
+	TLS TLS
 
-	// FollowRedirects will instruct the call to follow 301 redirects. Otherwise, the original 301 response
-	// is returned directly.
-	FollowRedirects bool
+	// Message to be sent.
+	Message string
```
Comment (howardjohn) [inline pkg/test/framework/components/echo/calloptions.go:112]: I think this is part of TCP

**Review by douglas-reid — APPROVED** [review]
Changes to E&T owned files LGTM.

File: pkg/test/framework/components/echo/calloptions.go [hunk 4]
```diff
@@ -24,6 +24,55 @@ import (
...
+
+// TCP settings
+type TCP struct {
+	// ExpectedResponse asserts this is in the response for TCP requests.
+	ExpectedResponse *wrappers.StringValue
+}
+
 // CallOptions defines options for calling a Endpoint.
 type CallOptions struct {
 	// Target instance of the call. Required.
```
Comment (howardjohn) [inline pkg/test/framework/components/echo/calloptions.go:78]: we may want to make port a non-pointer btw. That has been a minor inconvenience for a while

**Review by howardjohn — APPROVED** [review]

File: pkg/test/framework/components/echo/calloptions.go (hunk 3 above)
Comment (nmittler) [reply pkg/test/framework/components/echo/calloptions.go:117]: It seems to be used by multiple protocols

File: pkg/test/framework/components/echo/calloptions.go (hunk 2 above)
Comment (nmittler) [reply pkg/test/framework/components/echo/calloptions.go:78]: Yeah ... already raised https://github.com/istio/istio/issues/37685

**Pull request merged**